client/src/main/scala/spatutorial/client/components/FlightsWithSplitsTable.scala

```diff
@@ -190,55 +189,47 @@ object FlightsWithSplitsTable {
     else s"${max}C"
   }
 
-  def reactTableFlightsAsJsonDynamic(flights: FlightsWithSplits): List[js.Dynamic] = {
-
-    flights.flights.map(flightAndSplit => {
-      val f = flightAndSplit.apiFlight
-      val literal = js.Dynamic.literal
-      val splitsTuples: Map[PaxTypeAndQueue, Int] = flightAndSplit.splits
-        .splits.groupBy(split => PaxTypeAndQueue(split.passengerType, split.queueType)
-      ).map(x => (x._1, x._2.map(_.paxCount).sum))
-
-      import drt.shared.DeskAndPaxTypeCombinations._
-
-      val total = "API total"
-
-      def splitsField(fieldName: String, ptQ: PaxTypeAndQueue): (String, scalajs.js.Any) = {
-        fieldName -> (splitsTuples.get(ptQ) match {
-          case Some(v: Int) => Int.box(v)
-          case None => ""
-        })
-      }
-
-      def splitsValues = {
-        Seq(
-          splitsTuples.getOrElse(PaxTypesAndQueues.eeaMachineReadableToEGate, 0),
-          splitsTuples.getOrElse(PaxTypesAndQueues.eeaMachineReadableToDesk, 0),
-          splitsTuples.getOrElse(PaxTypesAndQueues.eeaNonMachineReadableToDesk, 0),
-          splitsTuples.getOrElse(PaxTypesAndQueues.visaNationalToDesk, 0),
-          splitsTuples.getOrElse(PaxTypesAndQueues.nonVisaNationalToDesk, 0)
-        )
-      }
+  val uniqueArrivalsWithCodeShares = CodeShares.uniqueArrivalsWithCodeshares((f: ApiFlightWithSplits) => identity(f.apiFlight)) _
 
-      literal(
-        //        "Operator" -> f.Operator,
-        "Timeline" -> "0",
-        "Status" -> f.Status,
-        "Sch" -> f.SchDT,
-        "Est" -> f.EstDT,
-        "Act" -> f.ActDT,
-        "Est Chox" -> f.EstChoxDT,
-        "Act Chox" -> f.ActChoxDT,
-        "Gate" -> f.Gate,
-        "Stand" -> f.Stand,
-        "Pax" -> paxOriginDisplay(f.MaxPax, f.ActPax, splitsTuples.values.sum),
-        "Splits" -> splitsValues.mkString("|"),
-        "TranPax" -> f.TranPax,
-        "Terminal" -> f.Terminal,
-        "ICAO" -> f.ICAO,
-        "Flight" -> f.IATA,
-        "Origin" -> f.Origin)
-    })
+  def reactTableFlightsAsJsonDynamic(flights: FlightsWithSplits): List[js.Dynamic] = {
+    val uniqueArrivals: List[(ApiFlightWithSplits, Set[ApiFlight])] = uniqueArrivalsWithCodeShares(flights.flights)
```

Comment:
what's the intended function of this change? how does it relate to the change of our table?